creator-node/test/TranscodeDelegationManager.test.js

```diff
@@ -0,0 +1,274 @@
+    assert.strictEqual(randomSPs.length, 3)
+    randomSPs.forEach((sp) => {
+      assert.ok(allSPsSet.has(sp))
+    })
+
+    // Mock libs call to only return 1 sp
+    libsMock.ethContracts.getServiceProviderList = () => {
+      return allSPs.slice(0, 1)
+    }
+
+    // Return 1 SP for libs
+    let oneSP = await TranscodeDelegationManager.selectRandomSPs(libsMock)
+    assert.strictEqual(oneSP.length, 1)
+    assert.ok(allSPsSet.has(oneSP[0]))
+  })
+
+  // handOffToSP()
+
+  it('When handing off transcode request to an sp, if the health check fails, throw error', async function () {
+    sinon
+      .stub(TranscodeDelegationManager, 'fetchHealthCheck')
+      .rejects(new Error('failed to fetch health check response'))
+
+    let resp
+    try {
+      resp = await TranscodeDelegationManager.handOffToSp(logger, {
+        sp: 'http://some_cn.com',
+        req: reqMock
+      })
+      assert.fail('Should not have passed')
+    } catch (e) {
+      assert.ok(e.message === 'failed to fetch health check response')
+    }
+  })
+
+  it('When handing off transcode request to an sp, if the health check passes but transcode req does not, throw error', async function () {
+    sinon.stub(TranscodeDelegationManager, 'fetchHealthCheck').resolves()
+
+    sinon
+      .stub(TranscodeDelegationManager, 'sendTranscodeAndSegmentRequest')
+      .rejects(new Error('failed to send transcode and segment request'))
+
+    let resp
+    try {
+      resp = await TranscodeDelegationManager.handOffToSp(logger, {
+        sp: 'http://some_cn.com',
+        req: reqMock
+      })
+      assert.fail('Should not have passed')
+    } catch (e) {
+      assert.ok(e.message === 'failed to send transcode and segment request')
+    }
+  })
+
+  it('When handing off transcode request to an sp, if both the health check and transcode passes, return uuid', async function () {
+    sinon.stub(TranscodeDelegationManager, 'fetchHealthCheck').resolves()
+
+    const expectedUuid = uuid.v4()
+    sinon
+      .stub(TranscodeDelegationManager, 'sendTranscodeAndSegmentRequest')
+      .resolves(expectedUuid)
+
+    try {
+      const actualUuid = await TranscodeDelegationManager.handOffToSp(logger, {
+        sp: 'http://some_cn.com',
+        req: reqMock
+      })
+      assert.strictEqual(actualUuid, expectedUuid)
+    } catch (e) {
+      assert.fail(
+        `Transcode handoff failed and UUID was not returned: ${e.message}`
+      )
+    }
+  })
+
+  it('When polling for transcode, if polling fails, throw error', async function () {
+    sinon.stub(TranscodeDelegationManager, 'fetchHealthCheck').resolves()
+
+    const expectedUuid = uuid.v4()
+    sinon
+      .stub(TranscodeDelegationManager, 'sendTranscodeAndSegmentRequest')
+      .resolves(expectedUuid)
+
+    sinon
+      .stub(TranscodeDelegationManager, 'asyncRetry')
+      .rejects(new Error('polling failed'))
+
+    try {
+      const actualUuid = await TranscodeDelegationManager.handOffToSp(logger, {
```

Comment:
i think maybe here we should be calling `TranscodeDelegationManager.handoff` the OG wrapper fn bc we're testing the functionality of the constituent pieces